adala/skills/collection/entity_extraction.py

```diff
@@ -270,22 +270,33 @@ def extract_indices(self, df):
             text = row[input_field_name]
             entities = row[output_field_name]
             to_remove = []
+            found_entities_ends = {}
             for entity in entities:
                 # TODO: current naive implementation assumes that the quote_string is unique in the text.
                 # this can be as a baseline for now
                 # and we can improve this to handle entities ambiguity (for example, requesting "prefix" in response model)
                 # as well as fuzzy pattern matching
+                ent_str = entity[self._quote_string_field_name]
+                # check the last index of the found entities, to start the search from
+                start_search_idx = 0
+                for found_ent in found_entities_ends:
+                    # to avoid overlapping entities
+                    if found_ent.startswith(ent_str):
+                        start_search_idx = found_entities_ends[found_ent]
+                        break
+
```

Comment:
```suggestion
                    # to avoid overlapping entities
                    start_search_idx = max(
                        [found_entities_end[found_end] for found_ent in found_entities_ends if found_end.startswith(ent_str)]
                    )

```
shouldn't we take the max end_index over all entities that match, not the first one found?